### PR TITLE
Allow mosviz load_data to load only 1d or 2d spectra

### PR DESCRIPTION
**Pull request by javerbukh**
### Description

This pull request is to address `mosviz.load_data()` not working if only spectra 1d or 2d are provided. Also, if valid values are not provided in `load_data`, the method should return after sending a Snackbar message.

EDIT: This is only implemented for NIRSpec data since it is the only instrument with separate loaders for 1d and 2d spectra.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -110,6 +110,8 @@ Mosviz
 
 - Data unassigned a row is hidden under the subdropdown in the data dropdown. [#1798, #1808]
 
+- Allow Mosviz `load_data` method to load only 1d or 2d spectra. [#1833]
```
Comment (pllim) [inline CHANGES.rst:113]:
```suggestion
- Allow Mosviz ``load_data`` method to load only 1D or 2D spectra. [#1833]
```

File: jdaviz/configs/mosviz/helper.py [hunk 2]
```diff
@@ -463,14 +464,26 @@ def load_data(self, spectra_1d=None, spectra_2d=None, images=None,
             self.load_metadata(spectra_2d, spectra=True)
             self.load_metadata(spectra_1d, spectra=True, sp1d=True, ids=spectra_1d)
 
+        elif spectra_1d:
```
Comment (pllim) [inline jdaviz/configs/mosviz/helper.py:467]: Do we have to worry about the combo of spectra_1d + images only?

File: jdaviz/configs/mosviz/helper.py [hunk 3]
```diff
@@ -463,14 +464,26 @@ def load_data(self, spectra_1d=None, spectra_2d=None, images=None,
...
+            self.load_1d_spectra(spectra_1d, spectra_1d_label)
+            allow_link_table = False
+
+        elif spectra_2d:
```
Comment (pllim) [inline jdaviz/configs/mosviz/helper.py:471]: Do we have to worry about the combo of spectra_2d + images only?

File: jdaviz/configs/mosviz/helper.py [hunk 4]
```diff
@@ -463,14 +464,26 @@ def load_data(self, spectra_1d=None, spectra_2d=None, images=None,
...
+
         else:
             msg = "Warning: Please set valid values for the load_data() method"
+            msg = SnackbarMessage(msg, color='warning', sender=self)
+            self.app.hub.broadcast(msg)
+            return
 
         if msg:
             msg = SnackbarMessage(msg, color='warning', sender=self)
             self.app.hub.broadcast(msg)
```
Comment (pllim) [inline jdaviz/configs/mosviz/helper.py:483]: This is no longer needed with your change above.
```suggestion
```

File: jdaviz/configs/mosviz/helper.py [hunk 5]
```diff
@@ -463,14 +464,26 @@ def load_data(self, spectra_1d=None, spectra_2d=None, images=None,
...
+            allow_link_table = False
+
+        elif spectra_2d:
+            self.load_2d_spectra(spectra_2d, spectra_2d_label)
+            allow_link_table = False
+
         else:
             msg = "Warning: Please set valid values for the load_data() method"
+            msg = SnackbarMessage(msg, color='warning', sender=self)
+            self.app.hub.broadcast(msg)
```
Comment (pllim) [inline jdaviz/configs/mosviz/helper.py:478]:
```suggestion
            self.app.hub.broadcast(SnackbarMessage(
                "Warning: Please set valid values for the load_data() method",
                color='warning', sender=self))
```

**Comment by javerbukh** [comment]
Thank you for the suggestions @pllim, should be ready for another look!

File: jdaviz/configs/mosviz/tests/test_data_loading.py [hunk 6]
```diff
@@ -128,6 +128,26 @@ def test_load_multi_image_spec(mosviz_helper, mos_image, spectrum1d, mos_spectru
...
 
 
+def test_load_multi_image_and_spec1d_only(mosviz_helper, mos_image, spectrum1d):
+    spectra1d = [spectrum1d] * 3
+    images = [mos_image] * 3
+
+    mosviz_helper.load_data(spectra_1d=spectra1d, images=images)
+
+    assert mosviz_helper.app.get_viewer("table-viewer").figure_widget.highlighted == 0
+    assert len(mosviz_helper.app.data_collection) == 7
```
Comment (pllim) [inline jdaviz/configs/mosviz/tests/test_data_loading.py:138]: How did 3 + 3 become 7 in the data collection?

**Review by pllim — APPROVED** [review]
Code looks like code.

But I don't understand subtleties of Mosviz, so extra pairs of eyes would help.

**Comment by pllim** [comment]
Change log has conflict now but that can be resolved after second approval?

**Review by duytnguyendtn — COMMENTED** [review]
Amazing how we started with requiring all three data products, and now we're this flexible. Thanks Jesse! One thing I found though...

1. I tested this by loading the 1D and 2D spectra from the NIRspec example notebook, which completed successfully
2. When I tried to load the NIRISS example, `Spectrum1D.read()` failed. I don't know if this is because the NIRISS data files contain multiple spectra per file.

Does anyone else see the NIRISS issue, and should that stop this PR from proceeding?

**Comment by javerbukh** [comment]
I should probably have been more clear in the description that this only applies to NIRSpec because NIRISS does not have separate loaders. Once that is possible with NIRISS, I can see this work being extended to that instrument as well.

**Comment by pllim** [comment]
@duytnguyendtn are you satisfied by Jesse's reply? If so, are you willing to approve this? Thanks!

**Review by duytnguyendtn — APPROVED** [review]
Works for me, thanks!

**Comment by pllim** [comment]
Auto backport failed again. @javerbukh , do you want to do a manual backport or should we not backport at all?

**Comment by javerbukh** [comment]
I have not done a manual backport yet, so that could be good practice. Is there documentation I can follow?

**Comment by pllim** [comment]
The bot was supposed to tell us but looks like it is misbehaving. I'll dig out an old post and maybe just dump the instructions in the dev docs... why not?

**Comment by pllim** [comment]
@javerbukh , example instruction for another PR: https://github.com/astropy/astropy/pull/13933#issuecomment-1297771382

**Comment by javerbukh** [comment]
@pllim , opened #1842 for the backport, hopefully everything looks correct there!

**Pull request merged**